compiler/generator_js.c

```diff
@@ -1278,38 +1278,29 @@ static void generate(struct generator * g, struct node * p) {
 
 static void generate_class_begin(struct generator * g) {
     if (g->options->js_esm) {
-        w(g, "// deno-lint-ignore-file~N"
-             "import ~P from './base-stemmer.mjs'~N"
-             "~N"
-             "/** @typedef {{ stemWord(word: string): string }} Stemmer */~N"
-             "~N"
-             "/** @type {{ new(): Stemmer }} */~N"
-             "~C ~n = function() {~+~N"
-             "~M~D base = new ~P();~N");
-    } else {
-        w(g, "/**@constructor*/~N"
-             "~D ~n = function() {~+~N"
-             "~M~C ~P = require('./base-stemmer.js');~N"
-             "~M~D base = new ~P();~N");
+        w(g, "import { ~P } from './base-stemmer.esm.js'~N");
+        write_newline(g);
     }
+    w(g, "/** @constructor */~N"
+         "~C ~n = function() {~+~N"
+         "~M~C base = new ~P();~N");
     write_newline(g);
 }
 
 static void generate_class_end(struct generator * g) {
-    w(g, "~N");
+    write_newline(g);
     w(g, "~M/**@return{string}*/~N");
     w(g, "~Mthis['stemWord'] = function(/**string*/word) {~+~N");
     w(g, "~Mbase.setCurrent(word);~N");
     w(g, "~Mthis.stem();~N");
     w(g, "~Mreturn base.getCurrent();~N");
     w(g, "~-~M};~N");
     w(g, "~-};~N");
+    write_newline(g);
     if (g->options->js_esm) {
-        w(g, "~N"
-             "export default ~n~N");
+        w(g, "export { ~n };~N");
     } else {
-        w(g, "~N"
-             "if (typeof module === 'object' && module.exports) module.exports = ~n;~N");
+        w(g, "globalThis.~n = ~n;~N");
```

Comment:
This `"globalThis.{LANG}Stemmer = {LANG}Stemmer` line is not strictly required, as `var` has global scope, but is included for explicitness.